### ISSUE TITLE
New BunkerSilo mode is out , please test it 

### ISSUE THREAD

**Issue by ThomasGaertner** [Courseplay/courseplay#1568]
I've just finished the first version of the BunkerSilo mode 10 . the filling function works fine so far, the leveling needs some adjustment. but first I need feedback from you...
[setup_mode10_course.pdf](https://github.com/Courseplay/courseplay/files/679279/setup_mode10_course.pdf)

**Comment by pseudex** [comment]
testing it right now with the silo on sosnovka next to the cows.
the description is a little bit misleading. It isn't really clear if you need a [P] waypoint inside the silo. I would call waypoint 1 just starting point.
I also got confused with the shield-error message. Took me a trip to the dealer until I found out, that the shield is fine. It just needs to go to the back :)

**Comment by pops64** [comment]
C:/Users/Ryan/Documents/My Games/FarmingSimulator2017/mods/FS17_Courseplay5/mode9.lua:561: attempt to index a nil value

I think I have it set up right. Mabye I don't. Thanks for the update
![farming simulator 17 12_30_2016 10_22_00 pm](https://cloud.githubusercontent.com/assets/10678220/21575569/66f01fa8-cede-11e6-812a-31ad07455b45.png)

**Comment by JDTiger** [comment]
Works quite well, the silo is nice and even, if the tractor drive slowly (usually i set it to like 5km/h), only thing i don't like, is the heap at the end of the silo:  
![fsscreen_2016_12_31_16_31_18](https://cloud.githubusercontent.com/assets/9157835/21578174/e97b0f4e-cf76-11e6-8f61-a013ba6ac94c.png)

**Comment by Paul2510** [comment]
pops64 you need chaff in the Silo

**Comment by pseudex** [comment]
@JDTiger I noticed the same with the heap. Which trailer did you use for unloading? The first time I used one with 50.000l volume. The bergmann with 34.000 seem to be a bit better because they don't reach the end of the bunker while unloading. You could also try to level it from both sides. This mode will even out the heap quite good.

Another thing I saw was, that on the right side there was chaff going on the outside (through the wall, abouve the wall, I don't know. bunker was at middle marker). When I changed the width of the tool to 3,7m it worked better.

There is still some tuning to do. But for the first version this mode is awesome. Because I wanted to get round this bumpy hills I used a Silo mod with load and unload trigger. Now thanks to you, I'm only doing bunkers!

**Comment by tjb8907** [comment]
I played with this a bunch last night. Have a few comments, I was testing at the BGA on Goldcrest Valley. The fill up version definitely works better, but I like the leveling mode too, it's takes awhile for that made to get it really smooth, but it is nice. I didn't let either of them fill completely either, ran out of time, but I got probably close to 800k in each mode. 

I noticed sometimes the Tractor will go to the wait point and the semi would then move to about where it starts reversing and stops and then the tractor would go in and make another pass then go park again. Not sure if that is intentional or if it's because the semi gets out of range. A waitpoint on the unload route we be nice, also would be cool if the tractor only stops semi's/tractors that have a unload point in the same bunker, so you could be dumping and leveling 2 bunkers at the same time. 

The reason I say that is because I had 3 trucking running the unload route and sometimes two or more would unload at all once without letting the tractor push. In fill up mode it actually handled it fairly well, but would be nice to be able to split it up, or have a way to let the tractor make 1 or 2 passes between trucks. 

Of course I understand this is very beginning stages, so probably need to focus on bugs instead of feature improvements, but just thought I'd share some of my thoughts and ideas. 

I'll try to add some screenshots later of some of the minor issues I had.

**Comment by ntrrlz** [comment]
I tried it in the bga in GCV. I had an unload course where the tractor drive trough the silo and then drives back to the enrance via the slurry pit. After the trailer is unloaded the leveler start again, but then the tractor with trailer drives back into range and the leveler stops to let the trailer unload again. Maybe you could add a check that the trailer is not empty and if it is empty the just let it continue.

Also it would be nice if courseplay would fold and unfold the leveler.

**Comment by SunkenL** [comment]
Looking forward to try this out when i have time. A question though.. It is possible to use with different sized/type of silos or just the standard format on the original maps?

**Comment by ThomasGaertner** [comment]
@SunkenL : it should work with all silos which use the standard Giants BunkerlSilo configuration (a area with a interaction trigger arround it )

**Comment by ThomasGaertner** [comment]
-fixed: empty silo -the tractor will go to wait till there is something in the silo
-added: filling up mode: the tractor will go to wait when all chaff is moved and then wait for new material
-not fixed atm: the heaps at the end when leveling, I will have to include some more calculations for it
-the emtpy trailer ignore stuff will be implementd soon
-what I forgot to mention: the  recongition radius is considered from the first way point , also called start point :-) and I based all the actions of tipping reverse into the silo. the tests with driving throught unloading failed hilariously ;-)

**Comment by rmn18** [comment]
I have an LUA error when the tractor with Schild in Mode 10 works. So Tractor I use the Fendt 900 with Holeras Shild. When the error comes then the tractor turns only in the circle. Log file I hang times times here.
[log.txt](https://github.com/Courseplay/courseplay/files/680214/log.txt)

**Comment by ThomasGaertner** [comment]
01.0003

**Comment by rmn18** [comment]
A thanks ,another LUA have sean in the log Error: Running LUA method 'update'.
C:/Users/Maiki/Documents/My Games/FarmingSimulator2017/mods/ZZZ_courseplay/mode10.lua:306: attempt to index field 'BunkerSiloMap' (a nil value) ,ok i attached the here 
[log.txt](https://github.com/Courseplay/courseplay/files/680220/log.txt)

**Comment by ThomasGaertner** [comment]
yea I suppose, you got stuck in traffic and stopped and restarted it inside the bunker :-)

**Comment by lcc1211** [comment]
Working well for me even on custom maps.  Really appreciate making this course.

**Comment by Nebular64** [comment]
Running 01.0004 and the fill function has been running great.  The leveling is quirky, but I know that will come in due time.  Great work!

**Comment by saku05** [comment]
Bonjour à tous
Vielen Dank für Ihre arbeit. Alles geht gut, kein problem mit diese version. 5.01.0001
https://www.youtube.com/watch?v=GdInMH8suBo

**Comment by pseudex** [comment]
@ThomasGaertner on waiting pos the bunker silo map is genereated every update run. at least it looks like that. frames are also dropping alot
https://youtu.be/BG0XLdfejIg

**Comment by Raints** [comment]
Not sure its a bug or not. But I did use AUTO blade height and every time tractor jumped then it did raise the blade by 0,05m. At the end I had big pile of chaff at start and tipper didnt tip anymore. Started use manual blade height and all went well untill somehow tractor lower the blade. Its not really big issue. It would be good if the tractor driving over a pile that is at the end, do not read it for the jump. Then he did not raise the blade so often.

**Comment by pops64** [comment]
@Raints Try filling up mode instead of leveling mode when unloading to silos

I have also noticed that occasionally the blade height will ride along the ground. Will do some testing to see if i can figure out how to reproduce. I have picture will upload later

**Comment by Raints** [comment]
@pops64  Thanks. I did try filling also, but I would like to get it compacted also same time. Atm all working, its weird because I started using version 001 again. In latest version my log was filled with somrthing called check for fill level or something like that. Sry, dont have that log anymore.
![fsscreen_2017_01_03_20_56_18](https://cloud.githubusercontent.com/assets/3618060/21619357/e83ba976-d1f7-11e6-9c7f-279f4dc0a621.png)

**Comment by pops64** [comment]
I have also noticed in unloading mode. When it tries driving off the heap some times it will drive right into a wall wont slip and just keep on going until it is out of the silo then do donuts. This might because instead of taking the center line out the silo it takes the edge and wedges it self along the wall.

**Comment by ThomasGaertner** [comment]
@pseudex : the FPS breakdown should now be history
@Raints : the shield height will be changed under certain conditions in automatic mode, i'm not satisfied with it yet and the prints are removed, now the log should stay clean....
@pops64 : the shield height is not considered in filling mode, the target height is set to 0 all time , but there is still room for improvement with some tractors...

**Comment by Paul2510** [comment]
HI! Könnte mann diesen Modus auch für für Radlader machen? Da ich am liebsten mit ihm schiebe da es mit diesen am besten planieren kann! Vielen Dank für diese tolle Idee und umsetzung!

**Comment by Raints** [comment]
Using version v5.01.00006 and all working just fine. I am testing new silo and all good so far. Have one question. As you can see in the picture the silo wont filled to the end. Maybe it will, no more time to test atm. Will do that later. 
![fsscreen_2017_01_04_10_07_43](https://cloud.githubusercontent.com/assets/3618060/21635452/7609a2d0-d266-11e6-968e-064f30532d95.png)

I know that CP in "NO MODS PHASE" atm, but when I disable DriveControl tractor with blade start acting weird (blade shakes up and down) If its enabled all stable. Maybe I have somekind of mod conflict. 

Keep up the great work.

**Comment by JDTiger** [comment]
Suggestion:
Integrate a mode to scan only for vehicle in CP mode 2
It would be nice to have this mode, so that one can have a wheelloader on mode 9 running to empty a silo, while filling up another one...

**Comment by gui7545** [comment]
Just exclude mode 9 because sprayers or liquid manure transporter will come refill at the BGA (mode 4 and mode 8 ?) and will disturb the mode 10

**Comment by JDTiger** [comment]
how do they disturb mode10? they drive around the silo, not through it...
The leveler/compactor has his scanning function, to clear the silo for vehicles delivering more chaff/gras...

**Comment by pops64** [comment]
Yea but you could just rely on the traffic collision detection system for all modes but 1 and 2 because 1,2 are the only ones interfering with mode 10

**Comment by SunkenL** [comment]
Just starting to try this out and it works well for me. But is there anyone else who have problem with the loading of the bunker? i try to back up with a load of grass but courseplay just won't dump it. Just drives away. Am i doing something wrong?
v.0011

**Comment by pops64** [comment]
Are you using a forage wagon? When CP reverse into the silo stop it before it goes forward and check to see if you can unload it manually. I know the forage wagons are  sometimes bugged in reverse may even extended to tippers haven't played tested enough.

**Comment by SunkenL** [comment]
Yes pops64 i used a foragewagon. Are in the process to make a route with the krone tipper instead. Will see if it works then..

**Comment by Raints** [comment]
I have a question in relation to the height of the blade. How is it calculated, from the ground or from the trigger. Why, I ask, I have a feeling that my edited map silos are too low and the tractor is stuck too often in filling mode.

**Comment by pops64** [comment]
ThomasGaertner said a while back
> the shield height is not considered in filling mode, the target height is set to 0 all time , but there is still room for improvement with some tractors... 

if it is getting stuck on the side that is cause of the work width.

**Comment by Raints** [comment]
@pops64    Probably my english not so good. What I mean is how the calculation works? Blade will hit the ground but driver get stuck. Work width is fine

**Comment by ntrrlz** [comment]
It looks to me that the blade height setting is not working correctly. When I set it at 0.7m or lower then it appears to be lower after a while. Also when I change the setting it behaves wrong. I had it set to 0.7m and I click on + and it jumps to 0.5m.
And I second issue I had. I had 2 tractors arriving at the same time and the first one stopped and waited for the second one and the second one was stuck in traffic behind the first one.

**Comment by JDTiger** [comment]
I can't use auto shield height anymore...
Is this a bug or a feature?

ATM it doesn't matter how i set teh shield, my silo always looks awkward really fast (waves in the silage) and the tractor will not push to the end of silo... wasn't a problem with auto shield height...

**Comment by ThomasGaertner** [comment]
@JDTiger : as I sadi, ther is room for improovement. atm I want to get rid of this 90 issues and then I will work further on it . 
@ntrrlz : i already have an idea how to solve this problem with two or more unoalding tractors, but didn't do it yet

**Comment by pops64** [comment]
Here is video of it running right into the wall for when you get time to come back to this for fine tuning.
[Link](https://www.dropbox.com/s/yzp8zlkgylgaeou/Farming%20Simulator%2017%201_6_2017%201_08_38%20PM.mp4?dl=0)

**Comment by AgroWillem** [comment]
Just a pratical question.
In the Fill Up mode:
What measures the working width? The width of the egaliser or the width of the bunker?
And how to obtain the best speed at pressing?

**Comment by pops64** [comment]
It is the width of the shield. To small it will hit the silo. To large it wont be able to reach some of the patches of silage.

**Comment by AgroWillem** [comment]
Ok. thx.
I don't know if this is an issue: in the FillUp mode the shield is leaving a layer of chaff on the bottom in front of the heap. It is not actually cleaning up the floor. Is this a setting somewhere?

**Comment by pops64** [comment]
No that is a bug. For some reason every once while when CP sets shield height to zero. It will set it to zero then hiccup and set it to some other height. If watch closely while it lowers the shield you will see it go down then jump back up.

**Comment by AgroWillem** [comment]
Yep indeed. Thx. It would probably fixed later on.

**Comment by IvoryBalboa** [comment]
Works quite well so far. But my problem is reverse tipping in the silo. The tipper reverses in the silo, starts unloading and continues his route while unloading. What am i doing wrong?

**Comment by AgroWillem** [comment]
I had i myself and figured out that a driven not deep enough backwards in the silo. Drive in reverse about half the length of your tractor into the silo. Then it works fine. ;)

**Comment by IvoryBalboa** [comment]
I drove even half the length of the whole silo while recording to check if that helps. But unfortunately it didnt help

**Comment by AgroWillem** [comment]
And did you drive in a straight line away from the silo?

**Comment by IvoryBalboa** [comment]
No i didnt. Now it works like a charm. Thanks man, that info saved my evening :)

**Comment by AgroWillem** [comment]
Your' welcome. Have a nice evening. ;)

**Comment by SnoopBok** [comment]
So happy with this mode, great job! I was wondering if there will be support in the future for backwards driving? Like the Fendt 900 series has and the Valtra S series.

**Comment by ThomasGaertner** [comment]
@SnoopBok : its on our looooong list ....

**Comment by rmn18** [comment]
What's this ?? Giants Bug ?? see image.
![fsscreen_2017_01_09_20_43_23](https://cloud.githubusercontent.com/assets/3426782/21780690/d9d309e2-d6ac-11e6-88ed-5677f663d9af.png)

**Comment by ThomasGaertner** [comment]
hi guys, with 0026 you will get the improved version of leveling and compacting
-driving-throught-unloading is now supported, it has to be set up in the HUD
-filling should now work with all tractors, they should find level0 by themself properly
-added support for more than one unloading tractors in range: the first one who gets in range will tip while the second one will wait till the first one is out of range. maybe I will need to add an possibility to clean up the area before the second one unloads. You schould test this.
- the automatic speed is hardcoded to 8km/h till the speed calculation stuff is ready
- and would a native english guy please check the grammar of the last inputs in translation_en.xml and fix it if its wrong , thanks

**Comment by pops64** [comment]
Thanks for the great update :D. You have a typo tho you spelled through as trought. One question filling up mode is for when the silo is almost full? This was not clear and I thought it meant something else. Thanks again.

**Comment by ThomasGaertner** [comment]
filling up mode is filling the silo from back to front with maximum silo height ,
leveling is making of layers till height x  (I tried till 1 000 000 fillLevel)...

**Comment by Raints** [comment]
All working good if I use reverse unloading and leveling mode. Did try drive through but leveler made high heap at the end and drivers keep getting stuck there,

Maybe there is some setup what is working better with this mode? Maybe tipper like KRÖGER HKD 302?

**Comment by pops64** [comment]
I also was getting this heap at the end plus a little pushed out of the silo. I don't think there is a better setup just fine tuning needed to be done

**Comment by ThomasGaertner** [comment]
which trailer did you use ? may I have screenshot and vehicles list ? thanks

**Comment by pops64** [comment]
I was using Farmtech TDK 1600 and the Krone Big X 580. I can test again with the same trailer but with a tractor. Ill upload screenshots in little bit.

**Comment by ThomasGaertner** [comment]
yea the shield setting up is intended to keep the silage in the silo. since you can move the chaff in only one direction, there will always be more material in the back than in the front and middle....

**Comment by pops64** [comment]
That's what I kinda of figured. My heap isn't unbearable just gets a little crazy. Is it possible to save the Silo is being field by to the vehicle so it doesn't reset ever load. Still working on more testing. Thanks

**Comment by lcc1211** [comment]
Except for the heap of silage at the back of the bunker, everything seems to be working just fine.

**Comment by AgroWillem** [comment]
It looks like that in both modes de chaff is deposited outside the back of the silo.
It is pushed too far too the end.
Look at the picture.
![fsscreen_2017_01_15_14_58_20](https://cloud.githubusercontent.com/assets/23076511/21963063/69dc0ee6-db33-11e6-9ba3-863371163cfe.png)

**Comment by lcc1211** [comment]
In the last version of CP with FS15, any chaff being left in the bunker was slowly deposited as the vehicle moved from one end of the bunker to the other.  Is that no longer possible with FS17?  It would seem to me if we could still do that then the leveling and compacting would be a little bit easier to accomplish.

**Comment by Satissis** [comment]
@Icc1211 I'm not totally sure I get what you mean by that ? Could you explain it a bit better ?

**Comment by JDTiger** [comment]
@Satissis  He means the old silage compacting behavior form FS15&13... where you level the silo simply be driving over it....

**Comment by lcc1211** [comment]
Once the tractor arrived at the bunker, it would slowly move from the entrance, through the bunker, to the exit and would start leaving chaff, but would sometimes temporarily stop depositing.  In other words the last of the chaff would be dropped just before exiting.  Have you a link for the last dev version of CP for FS 15?  If so I'll reinstall FS15 and create a course to demonstrate.

**Comment by pops64** [comment]
What Icc1211 is talking about in 15 if the speed through the silo was to fast it would toggle tipstate back and forth to evenly dump chaff. I don't think that is possible any more in 17 because of heaps any wear. The problem with the chaff spilling out is when you drive over a heap it squashes out. So maybe with drive through instead of dumping while driving stop at the same point as reverse, dump it, wait til it's done then proceed.(Or mabye keep it to the first half of the silo) This will also let you have more control over the leveling process and maybe create a ramp the whole length of the silo that way you don't have that heap at the which I have gotten a tractor stuck, luckily the shield had enough power to push it over. Just my two cents

**Comment by Karlshoj** [comment]
When the pusher goes forward after ending a pushing run it speeds to some 30 Kph, which I think is a bit too fast. Would it be possible to reduce to forward speed?

**Comment by Vanquish081** [comment]
When you unfold the blade, there are problem although the width indicated is the correct (5.5 mts.)

![Example](http://i.imgur.com/JL8Fn1y.jpg)

**Comment by ThomasGaertner** [comment]
@Karlshoj :it will be done soon
@Vanquish081: set it to 6m and it works

**Comment by Karlshoj** [comment]
Thomas,
Thanks for your reply.
It would be great if you could choose only to wait for vehicles bringing chaff/grass, that you could continue selling silage from bunker A at the same time as you fill bunker B ??

**Comment by Vanquish081** [comment]
For unload is the same route, but choose set unload point and no waypoint?

**Comment by Yasp0** [comment]
I have problems with a custom map. The tractors alignment to the bunker is off-center. It leaves space to one side and runs into the other side. Is this a map issue? How can I solve this?

**Comment by andreasthu** [comment]
so, i have testet the mode10 also.
yes its nerv when the shield bug out the chaff from the bunker silo area
but i play my own sosnovka-mod-map
i was able to stretch the silowall (collision and visual : x=1.03) see on the GE pic
the i made with the terrininfolayerpaintmode new groundcollis
(info layer painting - infolayer - tipCol) see on the GE pic,
who are a little bit in the bunkersiloarea (GE pic)

i testet it
Series9 driver to deliver chaff, return tipping in the silo (see pic)
MF  8700 with holaras5m (verteilen und verdichten)
all shield mods automatic, only 6m km/h speed and shield 5,5m broad
wonderfull
only by round 350000 in the silo i stopped and go to the backside of the silo and to the hill on the end manually with the MF and the shield in the Silo (den Hügel am ende manuell ins Silo zurückgeschoben)
after 700000 i repeat it.
but you can see, there is no silage outside the siloarea.
next i wil ltry the other shieldmod (aufschieben) 

lg
andreas

**Comment by AgroWillem** [comment]
The problem with filling the bunkers too far is gone.
It all looks very nice now.
![fsscreen_2017_01_22_14_44_28](https://cloud.githubusercontent.com/assets/23076511/22182943/369cde82-e0b2-11e6-852d-073118fb5e70.png)

**Comment by andreasthu** [comment]
And what have you changed?
I filled it yesterday with v 01.00038
And the traktor pushet the chaff out of the silo.
Only i make tip collis on the end no chaff was outside the silo.
Lg
Andreas

**Comment by AgroWillem** [comment]
I changed nothing. But now it works. Same map, same bunker, same equipment.
I thought the bug was now solved.
I use the FillUp mode with everything automatic. Compress speed 10km/u

**Comment by andreasthu** [comment]
ok thanks
i will trie the fillup mode as next.
lg
andreas

**Comment by KalasKonrad** [comment]
In my testing  the fillup mode dose not push too far. but the leveling mode in both reverse and drivethrough mode do. and i had this problem since the option of reverse or drivethrough unload.

**Comment by frm274** [comment]
Used Mode: FillUp
Problem: Tractor lowers the Shield, hits the Bunker, Slip goes to 100% because rear axle looses traction, Tractor goas back without reaching the final height. 
<a href="http://www.pic-upload.de/view-32549220/bunkerfill.jpg.html"><img src="http://www2.pic-upload.de/thumb/32549220/bunkerfill.jpg" alt="Bild hochladen" /></a>
Suggestion: If the tractor starts slipping, slowly lift the attacher...

**Comment by Vanquish081** [comment]
Is possible that the blade "jump" when start route for leveling and compacting in 00052?¿

**Comment by Blackmankot** [comment]
@pops64 Can this issue be closed? To cleanse the topic.

**Comment by pops64** [comment]
We are probably going to leave this open until the website is update with mode10

**Comment by Blackmankot** [comment]
@pops64 Good.

**Comment by StreelRodent** [comment]
Been trying to use this on the GreenRiver 2017 map, but there is so little space in front of the bunkers that the levelers desire to turn around now and then makes it crash into the walls. So basically whenever a load comes in I have to manually stop the tractor, pull it out, and then wait for the offload to start it again. Of course I also have a problem with courseplay not detecting the bunker when offloading the trailers, so they just dump it all at the front and then go into a limbo because they can't empty it all. Everything worked fine when I tested it with empty trailers, but now that it has to do it for real I got so many issues that courseplay isn't helping me at all. The combine mode also can't figure out how to keep up with the forager (Krone Big X 580) and keeps crashing the tractors into the trees or stopping them before the forager reaches its turn.
